syntax = "proto3";


message AuthRequest {
    string email = 1;
    string password = 2;
}

message AuthResponse {
    string token = 1;
}


message ValidateOrDestroyRequest {
    string token = 1;
}

message ValidateResponse {
    string sub = 1;
    string role = 2;
}

message DestroyResponse {
    string success = 1;
}


service Idm {

    rpc Authenticate(AuthRequest) returns (AuthResponse);

    rpc Validate(ValidateOrDestroyRequest) returns (ValidateResponse);

    rpc Destroy(ValidateOrDestroyRequest) returns (DestroyResponse);

}